src/nvim/path.c

```diff
@@ -2237,3 +2237,40 @@ int path_is_absolute_path(const char_u *fname)
   return *fname == '/' || *fname == '~';
 #endif
 }
+
+/// Builds a full path from an invocation name `argv0`, based on heuristics.
+///
+/// @param[in]  argv0     Name by which Nvim was invoked.
+/// @param[out] buf       Guessed full path to `argv0`.
+/// @param[in]  bufsize   Size of `buf`.
+///
+/// @see os_exepath
+void path_guess_exepath(char *argv0, char *buf, size_t bufsize)
+{
+  char *path = getenv("PATH");
+
+  if (path == NULL || path_is_absolute_path((char_u *)argv0)) {
+    xstrlcpy(buf, argv0, bufsize);
+  } else if (argv0[0] == '.' || strchr(argv0, PATHSEP)) {
+    if (os_dirname((char_u *)buf, MAXPATHL) != OK) {
+      buf[0] = NUL;
+    }
+    xstrlcat(buf, PATHSEPSTR, bufsize);
+    xstrlcat(buf, argv0, bufsize);
+  } else {
+    char  *pathbuf = xstrdup(path);
+    char  *runpath = xmalloc(MAXPATHL);
+    path = strtok(pathbuf, ENV_SEPSTR);
+    do {
+      strcpy(runpath, path);
+      xstrlcat(runpath, PATHSEPSTR, MAXPATHL);
+      xstrlcat(runpath, argv0, MAXPATHL);
+      if (os_can_exe((char_u *)runpath, NULL, false)) {
+        break;
+      }
+    } while ((path = strtok(NULL, ENV_SEPSTR)) != NULL);
+    xstrlcpy(buf, runpath, bufsize);
```

Comment:
Looks like if nothing found, it uses whatever was the last possible `nvim` location, regardless of whether that was found or not. Is this intentional?